utils/src/main/kotlin/Utils.kt

```diff
@@ -59,18 +58,21 @@ const val PARAMETER_ORDER_LOGGING = 2
 const val PARAMETER_ORDER_HELP = 100
 
 /**
- * Check whether the specified two or more collections have no elemets in common.
+ * Return the set of elements the specified two or more collections have in common, or an empty set if all collections
+ * are disjoint.
  */
-fun disjoint(c1: Collection<*>, c2: Collection<*>, vararg cN: Collection<*>): Boolean {
+fun <T> disjoint(c1: Collection<T>, c2: Collection<T>, vararg cN: Collection<T>): Set<T> {
```

Comment:
It now computes the intersection, how about renaming it to "intersect" ?

Reply:
I made the same mistake in my first iteration of this PR 😉 It does *not* compute the intersection of all collections, but it checks whether any two collections have a non-empty intersection. That's a difference! In that sense, the goal of the function still is to check whether all collections are disjoint.

Reply:
Oh alright, that's a different story then.

The function documentation in combination with the function name to me seems still misleading, maybe sth like the following makes it clear?

comment: "Return the set of elements which are contained in at least two of the given collections."
name: "allPairsDiscoint", "unitePairwiseIntersections" hm, finding a good name is quite tricky...

Reply:
I've improved the docs like you suggest, but I've kept the function name as it's still the same function as before, and I cannot think of a better name right now either 😉

Reply:
Thanks!